### PR TITLE
Mikado

### PR DESCRIPTION
**Pull request by lucventurini**
* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [x] This PR adds a new recipe.
* [x] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).

### CODE REVIEW

**Comment by lucventurini** [comment]
Hi @daler, unfortunately I am unable to configure Mikado correctly. This is due to some mis-configuration in the libmagic library in the Conda installation machines; when Mikado tries to interface with libmagic, it cannot find it, and crashes. I have added both "libmagic" and "python-magic" as dependencies (for building and running) so I do not presume that the problem stems from there.

Would you have any suggestion on how to configure the environment so that my tool is capable of finding the libmagic library?

**Comment by daler** [comment]
I don't see anything obvious, sorry. Seems to be a problem with the `python-magic` package in conda-forge, as the following recreates the error:

```bash
conda create -n magic python-magic
source activate magic
python -c 'import magic; magic.Magic()'
```

**Comment by lucventurini** [comment]
Hi @daler, finally Mikado builds correctly :-) Would you be able to officially accept it into the bioconda repository?
Also, I will have an updated version to load, version v1.1 - the version specified by this recipe is the one I am using for the academic paper, but development has since moved on.

Thanks again for your help!

File: recipes/mikado/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+{% set hash_type = "sha256" %}
+{% set hash_value = "2e668e098d465ef1bff837e152732b49bbb61f74cc5fbb938be9d72281bd710a" %}
+
+package:
+  name: '{{ name|lower }}'
+  version: '{{ version }}'
+
+source:
+  fn: '{{ version }}.{{ file_ext }}'
+  # url: https://files.pythonhosted.org/packages/f0/31/64f3630dfe4ed06b36226cf48d25370e8ee9ad875f0fb82e1dd2f129bdc2/Mikado-1.0.2.tar.gz
```
Comment (daler) [inline recipes/mikado/meta.yaml]: rm these comment lines please

File: recipes/mikado/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+  # url: file:////usr/users/ga002/venturil/workspace/mikado/dist/Mikado-1.0.2.tar.gz
+  url: https://github.com/lucventurini/mikado/archive/1.0.2.tar.gz
```
Comment (daler) [inline recipes/mikado/meta.yaml]: here you can use the jinja2 variable `{{ version }}` in place of `1.0.2` for easier future maintenance

File: recipes/mikado/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+  '{{ hash_type }}': '{{ hash_value }}'
+
+build:
+  number: 0
+  entry_points:
+    - mikado = Mikado:main
+    - daijin = Mikado.daijin:main
+  # noarch: python
```
Comment (daler) [inline recipes/mikado/meta.yaml]: rm the noarch line please

**Comment by daler** [comment]
@lucventurini Great! It looks like the solution was that somehow it was pulling in older versions of libmagic and python-magic? I'm not sure why that was happening but I'm glad it's working now.

I added some minor comments on the recipe, can you address them please? Then we can merge this, and then you can open another PR for v1.1.

**Comment by lucventurini** [comment]
Hi @daler, changes made as requested. I have also disabled the build for python 3.6 as this particular version has a bug triggered specifically with Python 3.6 (but not python 3.5). The updated version 1.1 and following will be available for python 3.6 as well.

How should I upload a new version of Mikado after this first official release is accepted? Should I just open a new pull request with the updated recipe?

Thank you again for your help.

**Comment by lucventurini** [comment]
And to answer your question, yes, the problem was an incompatibility between the conda-forge up-to-date libmagic and the bioconda python-magic library. If I may, might it be a good idea to remove those two libraries from bioconda and rely on the conda-forge versions?

**Comment by daler** [comment]
Thanks!

Now what we do is wait for the master branch to build. Once that's done, it will upload the bioconda package and v1.0.2 will be available, and it will continue to be available in the conda channel even as the recipe here changes. So you can now submit a PR that makes the changes to this recipe to update it to v1.1 (version, hash, py36).

As for python-libmagic, thanks for the info. I'll add the `broken` label to the bioconda versions of `libmagic` and `python-magic`. In your v1.1 recipe, can you remove the version pinning to test if it then correctly installs the conda-forge versions?

**Pull request merged**